.gitignore

```diff
@@ -20,6 +20,7 @@
 .pants.d
 .pants.run
 .pants.workdir.file_lock
+.lock.process_manager.*
```

Comment:
The proliferation of lockfiles and directories continues to be unfortunate. Can we put this under `.pids`?

Reply:
I thought there was a reason we kept all file locks top-level in the buildroot, but I forget why exactly. I suppose we could move it to `.pids`, but there's an ever so slight chance of racing on creation of `.pids` etc.

in this case, there's no proliferation tho - we're just trading the prior `$BUILDROOT/.pantsd.startup` lock for `$BUILDROOT/.lock.process_manager.pantsd` since pantsd is the only consumer of `ProcessManager.process_lock`. I just thought it'd be nice if in the future when we needed an interprocess lock for lifecycle etc to have that standardized for all `ProcessManager` instances without one-offs.

I've removed the `.pantsd.startup` from `.gitignore` to hopefully make this clearer.

Reply:
> I thought there was a reason we kept all file locks top-level in the buildroot, but I forget why exactly. I suppose we could move it to .pids, but there's an ever so slight chance of racing on creation of .pids etc.

IIRC, we don't keep them under `.pants.d` because of races with clean-all. But I can't think of a reason not to put them under `.pids`. Would be good to do that here if possible.

Reply:
k, done. there is a race with purging of the named pid metadata dirs like `.pids/pantsd` so we can't safely key to e.g. `.pids/pantsd/lock`, but afaict nothing purges the top level `.pids` dir, so nesting them top level in that dir should work similarly to nesting in the buildroot.

Reply:
Thanks!